src/Control/Effect/Catch/Internal.hs

```diff
@@ -0,0 +1,20 @@
+{-# LANGUAGE DeriveFunctor, ExistentialQuantification, StandaloneDeriving #-}
+module Control.Effect.Catch.Internal
```

Comment:
I had initially placed all of these in a single `Control.Effect.Internal` module, but giving them their own modules means that much less recompilation.